# HG changeset patch
# Parent 9b57a3298d77414d6949f03927254045fddc9a03
# User Ben Kelly <ben@wanderview.com>
Allow Cache to shutdown cleanly if no operations are ever performed.


diff --git a/dom/cache/Manager.cpp b/dom/cache/Manager.cpp
--- a/dom/cache/Manager.cpp
+++ b/dom/cache/Manager.cpp
@@ -1402,19 +1402,19 @@ Manager::Manager(const nsACString& aOrig
   } else {
     Shutdown();
   }
 }
 
 Manager::~Manager()
 {
   NS_ASSERT_OWNINGTHREAD(Manager);
+  Shutdown();
   Factory::Instance().Remove(this);
   if (mContext) {
-    mContext->CancelAll();
     mContext->ClearListener();
   }
   mIOThread->Shutdown();
 }
 
 Context*
 Manager::CurrentContext()
 {
